rct229/rulesets/ashrae9012019/section5/section5rule23.py

```diff
@@ -37,12 +28,16 @@ def __init__(self):
                 rmds_used=produce_ruleset_model_description(
                     USER=False, BASELINE_0=True, PROPOSED=True
                 ),
-                # Make sure surfaces are matched in SurfaceRule
                 list_path="$.building_segments[*].zones[*].surfaces[*]",
                 each_rule=Section5Rule23.BuildingRule.SurfaceRule(),
                 index_rmd=BASELINE_0,
             )
 
+        def list_filter(self, context_item, data):
+            surface_b = context_item.BASELINE_0
+            subsurfaces_b = surface_b.get("subsurfaces", [])
+            return len(subsurfaces_b) > 0
+
         class SurfaceRule(RuleDefinitionListIndexedBase):
```

Comment:
If the logic is removed from the surfaces, then you can directly pointing to the subsurface level.
set `list_path = "$.building_segments[*].zones[*].surfaces[*].subsurfaces[*]`

Reply:
A bit of conventions that goes back to early days is regardless whether there is any logics in building level or not, we will add building rule in order to attach the building id to the reporting json (calc_vals).
But for the rest, we usually skip the levels. You can keep the `SurfaceRule` for reporting purpose or remove it as I suggested. I will leave the decision to you.